### PR TITLE
[SG-35191] Accessibility: Announce updates on inputs that control live page state

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
### Description
1. Typing in this pattern input filters the number of results that are matched. It's difficult to determine if the user's input is actually filtering correctly or if there's a typo that matches nothing without navigating the page a significant amount.
Image
2. This is also true on the site-admin repo filters available on code intelligence policy edit pages.
Image

### Refs
[SourceGraph Issue](https://github.com/sourcegraph/sourcegraph/issues/35191)
[GitStart Issue](https://app.gitstart.com/clients/sourcegraph/tickets/SG-35191)

### Expected behavior
These should be announced as the user types, or at the very least we should have an auxiliary count attached to the form input as a hint or label.
### Test plan
1. Toggle the  `disabled` prop to `false` on the `ObjectMatchingGitPattern.tsx`.
2. Navigate to the `configuration` [here](https://k8s.sgdev.org/github.com/sourcegraph/go-imports-search/-/code-intelligence/configuration/Q29kZUludGVsbGlnZW5jZUNvbmZpZ3VyYXRpb25Qb2xpY3k6MQ==).
3. Use the `screen reader` and type in the `pattern input` and hear it announce your result.

## App preview:

- [Web](https://sg-web-contractors-sg-35191.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-gemywuwnqv.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/codeintel/configuration/components/GitObjectPreview.tsx [hunk 1]
```diff
@@ -93,7 +93,7 @@ const GitObjectPreview: FunctionComponent<React.PropsWithChildren<GitObjectPrevi
 }) => (
     <div>
         {GitObjectHeader}
-        <small>
+        <small aria-live="polite">
             {preview.preview.length === 0 ? (
                 <>Configuration policy does not match any known commits.</>
             ) : (
```
Comment (gitstart-sourcegraph) [inline client/web/src/enterprise/codeintel/configuration/components/GitObjectPreview.tsx:99]: Hi @umpox @limitedmage 
To ensure this works perfectly as expected (similar to use case in `RepositorPopover`), As referenced in image below WDYT about including the number of matches in text, 

image

Also to render `LoadingSpinner` here mutually exclusive to rendered text. i.e  moving this code block up here from below.

```
{previewLoading ? (
    <LoadingSpinner className={styles.loading} />
) : (
```

All these would help to ensure the content of `<small aria-live="polite">...</small>` mutates at every state to ensure they are appropriately announced  live on change.

Comment (umpox) [reply client/web/src/enterprise/codeintel/configuration/components/GitObjectPreview.tsx:99]: @gitstart-sourcegraph I think this is a good idea, please can you implement this?

Please also update this branch with the latest `main` and we'll get this merged!

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.04 kb) | 0.00% (+0.04 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/contractors/SG-35191/compare-report.html;tab=live_object) for a full comparison between the commits d34ffc9 and fd199d7 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/codeintel/configuration/components/GitObjectPreview.tsx (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/web/src/enterprise/codeintel/configuration/components/GitObjectPreview.tsx]: Thanks, updated now @umpox

**Review by umpox — APPROVED** [review]
👍

**Pull request merged**